### ISSUE TITLE
NullValue and EmptyValue are both applied.

### ISSUE THREAD

**Issue by mbudiu-vmw** [uniVocity/univocity-parsers#158]
I could not figure out whether this behavior is intended, but it was somewhat unexpected. 
In a writer I set the nullValue to the empty string, and the emptyValue to an empty pair of quotes.
The result was that null values were output as an empty pair of quotes.
This happens here:

```Java
class AbstractWriter {

    protected String getStringValue(Object element) {
        this.usingNullOrEmptyValue = false;
        if(element == null) {
            element = this.nullValue;
            this.usingNullOrEmptyValue = true;
            if(element == null) {
                return null;
            }
        }

        String string = String.valueOf(element);
        if(string.isEmpty()) {
            this.usingNullOrEmptyValue = true;
            return this.emptyValue;
        } else {
            return string;
        }
    }
}
```

I would have expected the first case to immediately return.

**Comment by jbax** [comment]
Can you please provide a test case that reproduces any unintended behavior? Thank you!

**Comment by mbudiu-vmw** [comment]
```Java
 @Test
    public void csvWriterTest() throws IOException {
        // The Csv writer we are using has a bug, reproduced with this test.
        String[] data = new String[]{ "", null };
        CsvWriterSettings settings = new CsvWriterSettings();
        CsvFormat format = new CsvFormat();
        settings.setFormat(format);
        settings.setEmptyValue("\"\"");
        settings.setNullValue("");
        Writer fw = new FileWriter("tmp");
        CsvWriter writer = new CsvWriter(fw, settings);
        writer.writeRow(data);
        writer.close();
        fw.close();
    }
```

This writes a file containing: `"", ""`, although I expected it to contain `"",`

**Comment by jbax** [comment]
Thank you for reporting this bug! Apparently it's been there since day 1 and I had to update a few very old test cases that were testing the wrong thing. I've released a 2.5.0-SNAPSHOT build to include the fix for this.
